app/tag/[category]/[title]/page.tsx

```diff
@@ -0,0 +1,70 @@
+import { fetchArticleByUUID } from "@/lib/fetchArticleByUUID";
+import { Metadata } from "next";
+import Image from "next/image";
+import styles from "./styles.module.css";
+import Link from "next/link";
+export async function generateMetadata({
+  params,
+}: {
+  params: { title: string };
+}): Promise<Metadata> {
+  var data = params.title.replaceAll("%", " ");
+  data = data.replaceAll("-", " ");
+  data = data[0].toUpperCase() + data.slice(1, data.length);
+
+  if (data) return { title: data, description: data };
+
+  return { title: "Not found", description: `${data} not found` };
+}
+
+async function page({ params }: { params: { title: string } }) {
```

Comment:
Yes. all server side components can be async so that I don't have to write async function in it again to fetch a data


`const article: Article = await fetchArticleByUUID(post_uuid);`